### PR TITLE
Fix lock rights checking

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #785

~~There is an `UNLOCK` right that can be set on many items in profile management, but it seems to not be really used.~~

~~I add some rights check on unlock form inside an asset.~~

~~To really fix this, new logic should be added on `CommonDBTM` (`canUnlock()` method and `UNLOCK` case in `can()` method) and in `CommonDBConnexity`.~~

### CODE REVIEW

**Comment by cedric-anne** [comment]
After looking better at the code, I think Irealized I confused Lock and ObjectLock.

So changes are:
 - tab display is now conitionned by parent item `UPDATE` right instead of `CREATE` (restoring / purging child elements is obviously related to an update of their parent),
 - ability to perform restore / purge on child elements is based on child elements `UPDATE` / `PURGE` rights.

File: front/lock.form.php [hunk 1]
```diff
@@ -39,19 +39,30 @@
 if (isset($_POST['itemtype'])) {
 
    $itemtype    = $_POST['itemtype'];
+   /* @var Computer $source_item */
```
Comment (trasher) [inline front/lock.form.php]: What's the purpose of that?

Comment (cedric-anne) [reply front/lock.form.php]: Code assist in IDE.

Comment (cedric-anne) [reply front/lock.form.php]: But it should not have been pushed here.

**Comment by trasher** [comment]
I did not pay attention it was targetting 9.3; it should rather tagret master

**Pull request merged**